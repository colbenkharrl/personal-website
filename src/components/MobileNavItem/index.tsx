import { Popover } from '@headlessui/react';
import Link from 'next/link';

export function MobileNavItem({
  href,
  children,
}: {
  href: string;
  children: React.ReactNode;
}) {
  return (
    <li>
      <Popover.Button
        as={Link}
        href={href}
        className="block py-2"
        data-testid="MobileNavItem"
      >
        {children}
      </Popover.Button>
    </li>
  );
}
