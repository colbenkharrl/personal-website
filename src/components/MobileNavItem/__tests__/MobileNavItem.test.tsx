import { render, screen } from '@testing-library/react';
import * as stories from '../__stories__/MobileNavItem.stories';
import { composeStories } from '@storybook/react';
import { Popover } from '@headlessui/react';

const { Basic } = composeStories(stories);

it('MobileNavItem matches snapshot (regression test)', () => {
  render(
    <Popover>
      <Basic />
    </Popover>,
  );
  expect(screen.getByTestId('MobileNavItem')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
