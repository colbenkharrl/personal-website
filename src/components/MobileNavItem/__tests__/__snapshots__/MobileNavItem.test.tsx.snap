// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavItem matches snapshot (regression test): Full component snapshot. 1`] = `
<a
  aria-expanded="false"
  class="block py-2"
  data-headlessui-state=""
  data-testid="MobileNavItem"
  href="https://www.example.com/"
  id="headlessui-popover-button-:r0:"
>
  Example Link
</a>
`;
